### PR TITLE
Add Google OR-tools deployer for Mac OS X

### PR DESCRIPTION
**Pull request by lolski**
## What is the goal of this PR?

We have added a deployer rule for us to deploy Google OR-tools for the OS X platform. The deployer rule is custom, since it is customised to deploy a ready made java library rather than a java library generated within Bazel.

## What are the changes implemented in this PR?

1. `WORKSPACE`: Imports Google OR-Tools for OS X `tar` archive and export the relevant binary files (jars, srcjars, javadocs, ,and poms)
2. `//library/ortools`: Create a deployer script to deploy the files onto `repo.grakn.ai`

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -70,3 +70,11 @@ rules_pkg_dependencies()
 
 load("//library/maven:rules.bzl", "maven")
 maven(["org.zeroturnaround:zt-exec", "com.eclipsesource.minimal-json:minimal-json"])
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
```
Comment (lolski) [inline WORKSPACE:74]: Clean `WORKSPACE`

File: library/ortools/BUILD [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        # ortools-java
+        "@ortools_osx//:pom-local.xml",
+        "@ortools_osx//:ortools-java-8.0.8283.jar",
+        "@ortools_osx//:ortools-java-8.0.8283-sources.jar",
+        "@ortools_osx//:ortools-java-8.0.8283-javadoc.jar",
+    ]
+)
+
+#sh_binary(
```
Comment (lolski) [inline library/ortools/BUILD:22]: delete this comment, as well as the script (`deploy.sh`)

File: library/ortools/Deploy.kt [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    val otJava_JavadocFile = Paths.get("external", "ortools_osx", "ortools-java-$otVersion-javadoc.jar")
+
+
+    deployMaven(otJava_PomFile, username, password, repository, otGroupId, otJava_ArtifactId, otVersion, "pom")
+    deployMaven(otJava_JarFile, username, password, repository, otGroupId, otJava_ArtifactId, otVersion,  "jar")
+    deployMaven(otJava_SrcJarFile, username, password, repository, otGroupId, otJava_ArtifactId, otVersion, "srcjar")
+    deployMaven(otJava_JavadocFile, username, password, repository, otGroupId, otJava_ArtifactId, otVersion, "javadoc")
+}
+
+fun deployMaven(source: Path, username: String, password: String, repository: String, groupId: String, artifactId: String, version: String, type: String) {
```
Comment (lolski) [inline library/ortools/Deploy.kt:61]: `private`

File: WORKSPACE [hunk 4]
```diff
@@ -68,5 +72,5 @@ rules_pkg()
 load("@rules_pkg//:deps.bzl", "rules_pkg_dependencies")
 rules_pkg_dependencies()
 
-load("//library/maven:rules.bzl", "maven")
+# Load Maven artifacts
```
Comment (flyingsilverfin) [inline WORKSPACE:75]: good!

File: library/ortools/Deploy.kt [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+    val repository = "https://repo.grakn.ai/repository/maven"
+    val otGroupId = "com/google/ortools"
+    val otVersion = "8.0.8283"
+
+    /*
+     * Google OT Darwin artifacts
+     */
+    val otDarwin_ArtifcatId = "ortools-darwin"
+    val otDarwin_PomFile = Paths.get("external", "google_ortools_osx", "pom-runtime.xml")
```
Comment (flyingsilverfin) [inline library/ortools/Deploy.kt:31]: hang on, i thought there's one JNI wrapper jar/pom/sources that are not platform dependent, and there's the set of ones specifically for darwin/osx... why are both of these listed as `_osx`?

Also, will darwin builds work for any mac? Is it better to keep darwin in the name to be obvious what version its expected to work (and upwards)?

**Review by flyingsilverfin — APPROVED** [review]

**Pull request merged**